DOCSUP-10286: Update Join Table Engine documentation (#27057)
ClickHouse/ClickHouse: docs/en/engines/table-engines/special/join.md

Enter `join_strictness` and `join_type` parameters without quotes, for example, `Join(ANY, LEFT, col1)`. They must match the `JOIN` operation that the table will be used for. If the parameters do not match, ClickHouse does not throw an exception and may return incorrect data.

## Specifics and Recommendations {#specifics-and-recommendations}

### Data Storage {#data-storage}

`Join` table data is always located in the RAM. When inserting rows into a table, ClickHouse writes data blocks to the directory on the disk so that they can be restored when the server restarts.

If the server restarts incorrectly, the data block on the disk might get lost or damaged. In this case, you may need to manually delete the file with damaged data.

### Selecting and Inserting Data {#selecting-and-inserting-data}

You can use `INSERT` queries to add data to the `Join`-engine tables. If the table was created with the `ANY` strictness, data for duplicate keys are ignored. With the `ALL` strictness, all rows are added.

> You cannot perform a `SELECT` query directly from the table. Instead, use one of the following methods:

Reviewer: It's not main use-case, but actually you can select data:

```
CREATE TABLE j (`a` String, `b` LowCardinality(String), `c` String) ENGINE = Join(ANY, LEFT, a);

insert into j values ('aa', 'vv', 'xx'), ('aa', '11', '22'), ('bb', 'AA', 'BB');

select * from j;

┌─a──┬─b──┬─c──┐
│ aa │ vv │ xx │
│ bb │ AA │ BB │
└────┴────┴────┘
```

```suggestion
Main use-cases for `Join`-engine tables are following:
```

-   Call the [joinGet](../../../sql-reference/functions/other-functions.md#joinget) function, which lets you extract data from the table the same way as from a dictionary.

### Deleting Data {#deleting-data}

`ALTER DELETE` queries for `Join`-engine tables are implemented as [mutations](../../../sql-reference/statements/alter/index.md#mutations). `DELETE` mutation reads filtered data and overwrites data of memory and disk.

### Limitations and Settings {#join-limitations-and-settings}

When creating a table, the following settings are applied:

-   [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls)
-   [max_rows_in_join](../../../operations/settings/query-complexity.md#settings-max_rows_in_join)
-   [max_bytes_in_join](../../../operations/settings/query-complexity.md#settings-max_bytes_in_join)
-   [join_overflow_mode](../../../operations/settings/query-complexity.md#settings-join_overflow_mode)
-   [join_any_take_last_row](../../../operations/settings/settings.md#settings-join_any_take_last_row)
-   [persistent](../../../operations/settings/settings.md#persistent)

The `Join`-engine tables can’t be used in `GLOBAL JOIN` operations.

> The `Join`-engine allows use [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls) setting in the `CREATE TABLE` statement. And [SELECT](../../../sql-reference/statements/select/index.md) query allows use `join_use_nulls` too. If you have different `join_use_nulls` settings, you can get an error joining table. It depends on kind of JOIN. When you use [joinGet](../../../sql-reference/functions/other-functions.md#joinget) function, you have to use the same `join_use_nulls` setting in `CRATE TABLE` and `SELECT` statements.

Reviewer: ```suggestion
The `Join`-engine allows to specify [join_use_nulls](../../../operations/settings/settings.md#join_use_nulls) setting in the `CREATE TABLE` statement. [SELECT](../../../sql-reference/statements/select/index.md) query should have the same `join_use_nulls` value.
```
